extended/src/test/java/apoc/ml/PromptIT.java

```diff
@@ -298,6 +299,58 @@ public void ragWithRelevantAttributesComparedToIrrelevantOneAndChatProcedure() {
                 });
     }
 
+    @Test
+    public void ragWithRelevantAttributesComparedToIrrelevantOneAndChatProcedureGpt4() {
```

Comment:
Visto che questo test è bello lungo, farei un metodo common tipo quello in seguito,
e poi questo test con gpt4 lo scriverei: 
```
    @Test
    public void ragWithRelevantAttributesComparedToIrrelevantOneAndChatProcedureGpt4() {
      ragWithRelevantAttributesCommon( Map.of(MODEL_CONF_KEY, model) ),
   }
```
mentre l'altro con:
```
ragWithRelevantAttributesCommon( Map.of() ),
```

---

METODO COMMON:

```
    private void ragWithRelevantAttributesCommon(Map<String, Object> config) {
        String question = "Which athletes won the gold medal in mixed doubles's curling  at the 2022 Winter Olympics?";

        // -- test with hallucinations, wrong winner names
        testCall(db, """
                CALL apoc.ml.openai.chat([
                    {role:"user", content: $question}
                ], $apiKey, $config)""", 
                map("apiKey", OPENAI_KEY, "question", question),
                r -> {
                    var result = (Map<String,Object>) r.get("value");

                    Map message = ((List<Map<String,Map>>) result.get("choices")).get(0).get("message");
                    assertEquals("assistant", message.get("role"));
                    String value = (String) message.get("content");

                    String msg = "Current value is: " + value;
                    assertTrue(msg, value.contains("gold medal"));
                    assertNot2022Winners(value);
                });

        // -- test RAG with irrilevant attributes
        testCall(db, QUERY_RAG,
                map("attributes", List.of("irrelevant", "irrelevant2"),
                        "question", "Which athletes won the gold medal in curling at the 2022 Winter Olympics?",
                        "conf", Util.merge( map(API_KEY_CONF, OPENAI_KEY), config )
                ),
                (r) -> {
                    String value = (String) r.get("value");
                    String message = "Current value is: " + value;
                    assertTrue(message, value.contains(UNKNOWN_ANSWER));

                    assertNot2022Winners(value);
                });

        // -- test RAG with relevant attributes
        testCall(db, QUERY_RAG,
                map(
                        "attributes", RAG_ATTRIBUTES,
                        "question", "Which athletes won the gold medal in curling at the 2022 Winter Olympics?",
                        "conf", Util.merge( map("apiKey", OPENAI_KEY), config )
                ),
                (r) -> {
                    String value = (String) r.get("value");
                    String message = "Current value is: " + value;
                    assertTrue(message, value.contains("Stefania Constantini"));
                    assertTrue(message, value.contains("Amos Mosaner"));
                    assertTrue(message, value.contains("Italy"));
                });
    }
```